lightningd/Makefile

```diff
@@ -132,9 +132,9 @@ LIGHTNINGD_COMMON_OBJS :=			\
 include wallet/Makefile
 
 # All together in one convenient var
-LIGHTNINGD_HEADERS = $(LIGHTNINGD_HEADERS_NOGEN) $(LIGHTNINGD_HEADERS_GEN) $(WALLET_LIB_HEADERS)
+LIGHTNINGD_HEADERS = $(LIGHTNINGD_HEADERS_NOGEN) $(LIGHTNINGD_HEADERS_GEN) $(WALLET_HDRS)
 
-$(LIGHTNINGD_OBJS): $(LIGHTNINGD_HEADERS) $(WALLET_HDRS)
+$(LIGHTNINGD_OBJS): $(LIGHTNINGD_HEADERS)
```

Comment:
Yech, this is a bigger mess.  LIGHTNINGD_HEADERS is defined twice in this Makefile.

We should include wallet/Makefile earlier, then append the WALLET_HDRS to LIGHTNINGD_HDRS.  (We should also have `$(WALLET_OBJS): $(LIGHTNINGD_HEADERS)` after that).

Reply:
Ok, updated to fix these other issues :)